src/main/java/net/testudobank/MvcController.java

```diff
@@ -806,8 +806,27 @@ public String sellCrypto(@ModelAttribute("user") User user) {
    */
   public String applyInterest(@ModelAttribute("user") User user) {
 
-    return "welcome";
-
+    // This method borrows much of its functionality from the withdraw() method
+    String userID = user.getUsername();
+    
+    // check that user is eligible for interest applied to their account (requirements
+    // are outlined in the Interest Feature Design document)
+    if (user.getBalance() == 0 || 
```

Comment:
These are conditions to verify if the interest rate can be applied to the account. If at least one of them is not satisfied, then "welcome" is returned.